src/Language/PureScript/AST/Declarations.hs

```diff
@@ -510,6 +510,10 @@ data Declaration
   --
   | TypeClassDeclaration SourceAnn (ProperName 'ClassName) [(Text, Maybe Kind)] [Constraint] [FunctionalDependency] [Declaration]
   -- |
+  -- A type class member declaration (type declaration, optional default implementation)
+  --
+  | TypeClassMemberDeclaration SourceAnn (Declaration, Maybe Declaration)
```

Comment:
I think it'd be good to use `ValueDeclaration` here, and take advantage of the existing desugaring code.

Reply:
The `(Declaration, Maybe Declaration)` is in fact `(TypeDeclarationData, Maybe ValueDeclarationData)`. Should I make this explicit?

Reply:
I'd prefer this be done like regular value and type declarations if possible.

Reply:
I was trying to prevent orphaned value declarations. I can see now that I should be doing that during desugaring.